source/base/tensor_product_polynomials.cc

```diff
@@ -84,23 +84,20 @@ TensorProductPolynomials<dim, PolynomialType>::compute_index(
   const unsigned int             i,
   std::array<unsigned int, dim> &indices) const
 {
-  Assert(i < Utilities::fixed_power<dim>(polynomials.size()),
-         ExcInternalError());
-  internal::compute_tensor_index(index_map[i],
-                                 polynomials.size(),
-                                 polynomials.size(),
-                                 indices);
-}
-
-
-
-template <>
-inline void
-TensorProductPolynomials<0, Polynomials::Polynomial<double>>::compute_index(
-  const unsigned int,
-  std::array<unsigned int, 0> &) const
-{
-  AssertThrow(false, ExcNotImplemented("This function does not work in 0-d!"));
+  if constexpr (dim == 0)
+    {
+      AssertThrow(false,
+                  ExcNotImplemented("This function does not work in 0-d!"));
```

Comment:
Any reason not to use `DEAL_II_NOT_IMPLEMENTED()` here...

Reply:
I just didn't pay enough attention. Fixed now.